Compile ulf

elafnm opened the pull request:
Greetings dears @playpaolo and @pooyan-dadvand ,

kindly please, find this pull request for the ULF application "with pybind",

thank you for your support and appreciated effort, 

best regards,

elafnm committed:
pybind

elafnm committed:
pybind

elafnm committed:
pybind

playpaolo committed:
Corrected the path for ULF_variables

elafnm committed:
pybind

playpaolo committed:
Added a test example dealing with bottle blowing

playpaolo committed:
Removed dependency on structural_application

and changed the matrices operations to the new format:
i.e. instead of: self.pA = self.space_utils.CreateEmptyMatrixPointer()
self.A = (self.pA).GetReference()

no we have:self.pA = self.space_utils.CreateEmptyMatrixPointer()
self.A = self.pA
(note that GetReference() is not defined for KratosCompressedMatrix type)

playpaolo committed:
added time integration scheme to ULF

elafnm committed:
pybind

elafnm committed:
Merge remote-tracking branch 'origin/master' into compile_ULF

x# the commit.

elafnm committed:
pybind

playpaolo committed:
Removed dependencies of PfemApplication

playpaolo committed:
Changed SolidMechanicsApplication  to StructuralMechanicsApplication

now we import StructuralMechanicsApplication for FSI cases (always use this one from now onwards). Corresponding test examples are added t othe ULFApplication

playpaolo committed:
Added an  FSI example

elafnm committed:
updating variables

elafnm committed:
updating variables

elafnm committed:
updating variables

elafnm committed:
updating variables

elafnm committed:
merging master

elafnm committed:
merging master branch

playpaolo committed:
Minor changes

playpaolo committed:
removed result files of test examples

playpaolo committed:
Merge remote-tracking branch 'refs/remotes/origin/compile_ULF'

Conflicts:
	applications/MeshingApplication/test_examples/dam3d_remesh.gid/out.out
	applications/ULFapplication/ULF_application.h

playpaolo committed:
removed results files

elafnm committed:
updated

elafnm committed:
Merge branch 'compile_ULF' of https://github.com/KratosMultiphysics/Kratos into compile_ULF

elafnm committed:
Merge branch 'master' of https://github.com/KratosMultiphysics/Kratos into compile_ULF

playpaolo committed:
Staging RVE application

playpaolo committed:
Merge branch 'master'

Conflicts:
	kratos/sources/variables.cpp

elafnm committed:
Merge branch 'compile_ULF' of https://github.com/KratosMultiphysics/Kratos into compile_ULF

playpaolo committed:
Changing to new format of adding processes to python:

FOR ALL PROCESSES... For example:    class_<PressureCalculateProcess, PressureCalculateProcess::Pointer, Process > etc
instead of 
class_<PressureCalculateProcess, Process > (as it was before)

elafnm committed:
updating variables

elafnm committed:
updated

rubenzorrilla commented on applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py:
```diff
@@ -234,7 +234,7 @@ def ImportModelPart(self):
 
         ## Sets DENSITY, VISCOSITY and SOUND_VELOCITY
-        self._set_physical_properties()
+        #self._set_physical_properties()
```
Why are you changing this solver?

playpaolo replied:
Who did these changes? Is it me or @elafnm ?

elafnm replied:
Greetings dear @playpaolo and @rubenzorrilla 

I am not sure honestly, maybe I modified it by mistake. I will revert this change when I will be adding/pushing all my changes.

best,

rubenzorrilla commented on .gitignore:
```diff
@@ -63,7 +63,20 @@ kratos/includes/kratos_version.h
 
 applications/FluidDynamicsApplication/test_examples/fs_cavity/run_test\.py
-
+applications/ULFapplication/test_examples/TRIALS/
```
You should edit your local ignore file as explained in https://github.com/KratosMultiphysics/Kratos/wiki/How-to-make-Git-ignore-the-files-resulting-from-a-compilation-without-conflicts-in-.gitignore

elafnm replied:
greetings dear,

done,

best regards,

rubenzorrilla commented on .gitignore:
```diff
@@ -63,7 +63,20 @@ kratos/includes/kratos_version.h
 applications/FluidDynamicsApplication/test_examples/fs_cavity/ProjectParameters\.py
```
Indeed, I'm wondering why this is already here...

rubenzorrilla commented on .gitignore:
```diff
@@ -63,7 +63,20 @@ kratos/includes/kratos_version.h
 applications/FluidDynamicsApplication/test_examples/fs_cavity/ProjectParameters\.py
 
 applications/FluidDynamicsApplication/test_examples/fs_cavity/run_test\.py
```
Same...

philbucher replied:
We should remove this ...

playpaolo replied:
None of these work I think. I mean these test examples.

rubenzorrilla commented on applications/MeshingApplication/custom_python/add_meshers_to_python.cpp:
```diff
@@ -157,7 +157,7 @@ void TriRegenerateMeshGLASS(TriGenGLASSModeler& Mesher, char* ElementName, char*
 void TriRegenerateMeshDROPLET(TriGenDropletModeler& Mesher, char* ElementName, char* ConditionName, ModelPart& model_part,NodeEraseProcess& node_erase, bool rem_nodes, bool add_nodes, double alpha_shape, double h_factor )
 {
-    Mesher.ReGenerateMesh(model_part,
+    Mesher.ReGenerateMeshDROPLET(model_part,
```
Follow the naming convention ReGenerateMeshDroplet.

rubenzorrilla commented on applications/MeshingApplication/test_examples/dam3d_remesh.gid/out.out:
```diff
@@ -1,5 +1,5 @@
 Applications Available:
-Import_MeshMovingApplication: False
+Import_ALEApplication: False
```
ALEApplication doesn't exist anymore. It has been renamed to MeshMovingApplication.

elafnm replied:
greetings dear,

the file was modified as per the master,

loumalouomega commented on .gitignore:
```diff
@@ -63,7 +63,20 @@ kratos/includes/kratos_version.h
 
 applications/FluidDynamicsApplication/test_examples/fs_cavity/run_test\.py
-
+applications/ULFapplication/test_examples/TRIALS/
```
You can not modify gitignore, please use the .git/info/exclude instead-> is personal

elafnm replied:
done dear,

loumalouomega reviewed (changes requested):
This is not in conditions for merging, right @roigcarlo ?

philbucher commented on applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py:
```diff
@@ -234,7 +234,7 @@ def ImportModelPart(self):
 
         ## Sets DENSITY, VISCOSITY and SOUND_VELOCITY
-        self._set_physical_properties()
+        #self._set_physical_properties()
```
You cannot do this, this breaks the fluid!

loumalouomega commented on applications/ExternalSolversApplication/custom_external_libraries/SuperLU_4.3/FORTRAN/f77exm.out:
```diff
@@ -1,9 +0,0 @@
-No of nonzeros in factor L = 835
```
Why do you remove this?

elafnm replied:
greetings dear,

the file was added as per the master,

elafnm replied:
Greetings dears @loumalouomega, @rubenzorrilla, and @philbucher,

kindly please, we would be very pleased and would highly appreciate your usual support by looking at the recommended changes,

Thanks and best regards,

loumalouomega replied:
I still don't understand why do you change this file

elafnm replied:
Greetings der @loumalouomega,

honestly dear, I am not sure if I changed it, 

dear @playpaolo, do you have an idea please about this file and if it was changed by you!

thanks and best regards,

loumalouomega replied:
This looks like a minor change, just curious, check my comments for the rest of the review

philbucher commented on applications/MeshingApplication/custom_python/add_meshers_to_python.cpp:
```diff
@@ -257,7 +257,7 @@ void  AddMeshersToPython(pybind11::module& m)
     .def(init< >())
     .def("ReGenerateMeshDROPLET",TriRegenerateMeshDROPLET)
-    .def("ReGenerateMeshDROPLET",&TriGenDropletModeler::ReGenerateMesh)
+    .def("ReGenerateMeshDROPLET",&TriGenDropletModeler::ReGenerateMeshDROPLET)
```
please see the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide) for how to choose names => should be `ReGenerateMeshDroplet`

philbucher replied:
@playpaolo this is still remaining

rubenzorrilla commented on kratos/includes/variables.h:
```diff
@@ -412,6 +412,10 @@ namespace Kratos
     KRATOS_DEFINE_VARIABLE( Matrix, INERTIA )
+    
+    //for ULF (surface_tension) application:
+    KRATOS_DEFINE_VARIABLE(double, TRIPLE_POINT)
```
Just asking, If this variables are related to the ULF application, why are you adding them in the core variables?

elafnm replied:
Greetings dear @rubenzorrilla,

I had these variables to be used in the meshing application, and I found some issues during compiling kratos when I defined them locally. Accordingly and it was advised by Mr. @roigcarlo that:
"If you need that variable to be used in the meshing we should move the definition to the core"

regards,

philbucher replied:
@elafnm this was like this until two months ago

Now you can define the same variable in several applications
i.e. you can now define the variables you need in the MeshingApp also there and then use them there

roigcarlo replied:
Yup, now you can leave them in the application

philbucher commented on kratos/python/add_containers_to_python.cpp:
```diff
@@ -644,6 +644,11 @@ void  AddContainersToPython(pybind11::module& m)
 
 
+    //for ULF (surface_tension) application:
+    KRATOS_REGISTER_IN_PYTHON_VARIABLE(m,TRIPLE_POINT)
```
Why do you add these variables to the Core?

They are only used in your application from what I can see

philbucher replied:
@playpaolo this was also changed at some point

philbucher commented on applications/applications_interface.py:
```diff
@@ -135,6 +138,15 @@ def ImportApplications(kernel, applications_path=application_directory):
 
+    if(Import_FluidRveLagrangeMultipliersApplication):
+        print("importing KratosFluidRveLagrangeMultipliersApplication ...")
+        sys.path.append(applications_path + '/fluid_rve_lagrange_multipliers_application/python_scripts')
```
Naming conventions for applications is UpperCamleCase

Btw I would recommend to make a separate Pull request for adding a new application, this title of this PR is misleading

playpaolo replied:
@philbucher so do you suggest to put these files in the stash and do the revert?

philbucher replied:
I guess I would create a branch from this one and revert all unwanted changes such that the new Branch contains only the `FluidRveLagrangeMultipliersApplication`

Not the expert in git though, there might be better options

elafnm replied:
greetings dear,

 the fluid_rve application was removed from the branch by Dr. Pavel,

best regards,

philbucher reviewed (changes requested):
See the comments

juliobarna committed:
new element based on a semi-explicit methodology for free surface flows

playpaolo committed:
Removing the fluid_rve application from the branch

Removing the fluid_rve application from the branch. Will be added in a separate branch

playpaolo committed:
Merged branch compile_ULF into compile_ULF

playpaolo committed:
removed fluid_rve 

want to keep it only in my machine

playpaolo committed:
clean-up

removed repeated lines, extra spaces etc-

loumalouomega commented on applications/ULFapplication/custom_processes/calculate_curvature.h:
```diff
@@ -267,7 +272,8 @@ namespace Kratos
 	      }
 	      
-	      if(im->FastGetSolutionStepValue(IS_STRUCTURE) != 0.0 && im->FastGetSolutionStepValue(TRIPLE_POINT)*1.0E8 == 0.0)
+ 	      if(im->FastGetSolutionStepValue(IS_STRUCTURE) != 0.0 && im->FastGetSolutionStepValue(TRIPLE_POINT)*1.0E8 == 0.0)
```
I recommend to use flags, is faster and you don't need an extra variable

elafnm replied:
Greetings dear @loumalouomega and very good day to you,

I would like to get your kind recommendation regarding using/defining flags locally,

As far as I am concerned, in our first pull request, when we were adding this file, https://github.com/KratosMultiphysics/Kratos/pull/1410, a similar flag issue was raised, and at that time it was not convenient to define our new flags in the core and I received the following feedback accordingly:

"If you don't think they are convenient for you, then just keep the current implementation. I also agree that adding a new local flag is not a good solution in this case"

so my concern here please, in the case now the flag variables can be defined locally, would you kindly please recommend me to follow a specific example/standard where these flags were defined and used locally.

thanks and best regards,

loumalouomega replied:
The list of flags you can use is here:

https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/kratos_flags.h

If you need a new one you need to propose in an independent PR. The flags are limited to 64 and we can not add them happily. The flags you can use, according to you code could be:

FLUID
BOUNDARY
INTERFACE
FREE_SURFACE

loumalouomega replied:
The only one you need to be careful is with ACTIVE. The nodes and conditions set as non ACTIVE will not be computed in the core builder and solvers

loumalouomega commented on applications/ULFapplication/custom_processes/find_nodal_h_process.h:
```diff
@@ -0,0 +1,354 @@
+/*
```
Please, use the one in the core. We are strongly against code duplication

loumalouomega commented on applications/ULFapplication/custom_processes/find_triple_point.h:
```diff
@@ -105,117 +108,40 @@ namespace Kratos
-		    is_free += neighb[i].FastGetSolutionStepValue(IS_FREE_SURFACE);
-		    is_struct += neighb[i].FastGetSolutionStepValue(IS_STRUCTURE);
-		    is_lagin += neighb[i].FastGetSolutionStepValue(IS_LAGRANGIAN_INLET);
+		    if (neighb[i].FastGetSolutionStepValue(IS_FREE_SURFACE) != 0.0)
```
Please use flags instead

elafnm replied:
Greeting dear @loumalouomega and very good day to you,

I just discussed this item with Dr. Alex, we can use 

KRATOS_CREATE_FLAG(STRUCTURE, 63);
 
but we need two more variables one for the triple points, and one for the Lagrangian-inlet

"If you need a new one you need to propose in an independent PR. "
so shall we go ahead with this pull request as is so that the code will be compiled, and then we start working on it in an independent PR!

thanks and best regard,

loumalouomega replied:
There is already one called INLET, the number of flags is limited and maybe we can not have all the ones one would like

elafnm replied:
thanks dear, so we will be needing one for the "triple_point".

best regards,

elafnm replied:
shall we go ahead with this pull request and then create a new pull request, 

or shall we include it in this pull request.

best regards,

loumalouomega replied:
Yes, change the ones that you can currently change with the existing flags

playpaolo replied:
@loumalouomega I am strongly against doing this in this pull request. Flags you are talking about involve changes in 75% of our application and also the meshers we use from MeshingApplication. I suggest to make a separate pull request for "moving" all our old flag variables (IS_FREE_SURFACE, IS_STRUCTURE, IS_TRIPPLE_POINT) to new format. But lets do it in a separate pull request, otherwise things will get mixed.

loumalouomega replied:
I understand, I have approved

philbucher replied:
I agree with not using flags in this PR

nevertheless I recommend to NOT compare doubles with `==` but rather sth like
`if abs(val1-val2) < 1e-15` (just an example)

Quoting https://floating-point-gui.de/errors/comparison/:
>Due to rounding errors, most floating-point numbers end up being slightly imprecise. As long as this imprecision stays small, it can usually be ignored. However, it also means that numbers expected to be equal (e.g. when calculating the same result through different correct methods) often differ slightly, and a simple equality test fails

I will remove my blocker since my concerns were addressed

loumalouomega commented on applications/ULFapplication/custom_python/add_custom_utilities_to_python.cpp:
```diff
@@ -52,16 +52,21 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 
 // Project includes
+#include <pybind11/pybind11.h>
```
I am quite sure many of these includes are not necessary

loumalouomega commented on applications/ULFapplication/python_scripts/SurfaceTension_monolithic_solver.py:
```diff
@@ -4,9 +4,23 @@
 from KratosMultiphysics.FluidDynamicsApplication import *
```
This solver is nor following the others applications "standard"

loumalouomega commented on applications/ULFapplication/python_scripts/SurfaceTension_monolithic_solver.py:
```diff
@@ -328,23 +350,35 @@ def FindNeighbours(self):
+        time = self.model_part.ProcessInfo.GetValue(TIME)
+        dt   = self.model_part.ProcessInfo.GetValue(DELTA_TIME)
         ################### For sessile drop examples
         for node in self.model_part.Nodes:
```
Use the VariableUtils(), is faster and more efficient

loumalouomega commented on applications/ULFapplication/python_scripts/runge_kutta_frac_step_solver.py:
```diff
@@ -0,0 +1,225 @@
+
+CheckForPreviousImport()
+
+def AddVariables(model_part):
```
This is not following the standard

loumalouomega commented on applications/ULFapplication/python_scripts/runge_kutta_frac_step_solver.py:
```diff
@@ -0,0 +1,225 @@
+
+
+def AddDofs(model_part):
+    for node in model_part.Nodes:
```
Use VariableUtils()

loumalouomega commented on applications/ULFapplication/python_scripts/runge_kutta_frac_step_solver.py:
```diff
@@ -0,0 +1,225 @@
+
+        (self.neigh_finder).Execute()
+        
+        for node in (self.model_part).Nodes:
```
This is super inefficient, please use VariableUtils()

loumalouomega commented on applications/ULFapplication/python_scripts/runge_kutta_frac_step_solver.py:
```diff
@@ -0,0 +1,225 @@
+
+
+        if(self.domain_size == 2):
+            box_corner1[0]=-0.0001
```
many hardcoded values, how do I know what it is the meaning of these?, please use Parameters

playpaolo replied:
Thanks. We will change it. @juliobarna  could you please take care of this one?

loumalouomega commented on applications/ULFapplication/python_scripts/ulf_strategy_PGLASS.py:
```diff
@@ -37,9 +37,9 @@ def __init__(self, model_part, time_scheme, linear_solver, convergence_criteria,
         self.pDx = self.space_utils.CreateEmptyVectorPointer()
         self.pb = self.space_utils.CreateEmptyVectorPointer()
```
Strategies should be in cpp

playpaolo replied:
@loumalouomega I think we are having some sort of conceptual misunderstanding here. We are not planning to change the structure of the entire ULFApplication in this pull request (when the application was originally added we were not asked to reproduce the format of say FluidDynamicsApplication or other "user-oriented" standardized application). All the python solvers in ULF are divided into two parts: solver and strategy. We are neither interested in switching to JSON at the moment as we are using a different problem type.

loumalouomega commented on applications/ULFapplication/test_examples/bottle_blowing/ProjectParameters.py:
```diff
@@ -0,0 +1,73 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+domain_size = 3
+
+
```
ProjectParameters should be in *.json format

elafnm replied:
Greetings dear @loumalouomega,

Thank you very much for your valuable and appreciated feedback, 

I will be discussing it with Dr. Pavel and do the needed changes accordingly,

Thanks and best regards,

loumalouomega replied:
Best regards

playpaolo replied:
Hi @loumalouomega ! The problemtype used in ULF application is not preparing json files. We can think of updating this, but not within this pull request. Note that in this pull request new functionality is added. We will work on the format in the future.

loumalouomega reviewed (changes requested):
Much better, but still some work required

loumalouomega commented:
OK, then I will approve, but wait for the approval of the others

El jue., 12 jul. 2018 8:39, Pavel Ryzhakov <notifications@github.com>
escribió:

> *@playpaolo* commented on this pull request.
> ------------------------------
>
> In applications/ULFapplication/python_scripts/ulf_strategy_PGLASS.py
> <https://github.com/KratosMultiphysics/Kratos/pull/2215#discussion_r201923718>
> :
>
> > @@ -37,9 +37,9 @@ def __init__(self, model_part, time_scheme, linear_solver, convergence_criteria,
>          self.pDx = self.space_utils.CreateEmptyVectorPointer()
>          self.pb = self.space_utils.CreateEmptyVectorPointer()
>
> @loumalouomega <https://github.com/loumalouomega> I think we are having
> some sort of conceptual misunderstanding here. We are not planning to
> change the structure of the entire ULFApplication in this pull request
> (when the application was originally added we were not asked to reproduce
> the format of say FluidDynamicsApplication or other "user-oriented"
> standardized application). All the python solvers in ULF are divided into
> two parts: solver and strategy. We are neither interested in switching to
> JSON at the moment as we are using a different problem type.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/2215#discussion_r201923718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEMgOaI7XHMubiPgbTy9EgZMX_DWusYks5uFu8LgaJpZM4UR_qr>
> .
>

loumalouomega reviewed (dismissed):
Let for the approval of the others

playpaolo committed:
Adding a hypoelastic element

for monolithic ALE fsi formulation

philbucher reviewed (dismissed):
Actually the changes I requested initially are again there, even though I think they were solved at some point 
e.g. changes to gitignore, the change in the fluid solver and adding the variables in the core instead of the application

@elafnm @playpaolo can you check please?

playpaolo committed:
Change of process name

Renamed ulf_apply_bc_process to mark_free_surface_process (more adequate to what the process is doing).

playpaolo committed:
Clean-up

removed "hard-cored" constant values..

playpaolo committed:
removing process

already existing in the core. avoiding code duplictaion

playpaolo committed:
minor cleanup

playpaolo commented:
Hi @philbucher and @loumalouomega ! One question..  How can we bring in "standard" .gitignore to our branch in case by mistake we pushed a modified .gitignore at some point?

philbucher commented:
I think in this case it would be the easiest to copy paste the one from master and commit it, then the changes should disappear

loumalouomega commented:
Indeed @philbucher

loumalouomega committed:
Merge branch 'master' into compile_ULF

playpaolo committed:
preparing to put the clean ones from master

playpaolo committed:
brought the clean ones from master

playpaolo commented:
@philbucher and @loumalouomega  could you please check if it is fixed now? thank you!

philbucher commented on .gitignore:
```diff
@@ -66,5 +66,3 @@ kratos/includes/kratos_version.h
 *.h~
 *.cpp~
-
-applications/python_scripts/beam_profiles\.csv
```
@playpaolo please don't remove this

philbucher reviewed (commented):
@playpaolo the three things I pointed out are still missing

playpaolo committed:
added this

-
-applications/python_scripts/beam_profiles\.csv

playpaolo committed:
Changed to match style

DROPLET->Droplet

playpaolo commented:
@elafnm could you please see the thing about the variables @philbucher  is talking about? please put all your variables in the application and not in the core of kratos.

loumalouomega committed:
Merge branch 'master' into compile_ULF

loumalouomega committed:
Reverting conflicts

loumalouomega committed:
Restoring file

loumalouomega commented:
I reverted last problematic files, approving

loumalouomega reviewed (dismissed):
Done

playpaolo commented:
Thank you @loumalouomega !

philbucher commented:
now from my side only the moving of the variables from the core to the application is left

elafnm committed:
Update variables.h

removing the variables from the core

elafnm committed:
Update variables.cpp

removing the surface tension variables from the core

elafnm committed:
Update add_containers_to_python.cpp

removing surface tension variables from core

elafnm committed:
Update ULF_application.cpp

elafnm committed:
Update ULF_application_variables.h

elafnm committed:
Update ULF_application_variables.cpp

elafnm committed:
Update kratos_ulf_python_application.cpp

elafnm committed:
Update meshing_application.h

elafnm committed:
Update meshing_application.cpp

elafnm committed:
Update kratos_meshing_python_application.cpp

elafnm commented:
greetings dears @playpaolo and @philbucher and very good day to you,

done, they were moved from the core and defined locally. Please let me know if you have any further concern and it will be my pleasure to correct it accordingly.

Thanks and best regards,

elafnm commented:
greetings dears @philbucher and @rubenzorrilla and very good day to you,

could you please have a look at the changes done as per your recommendations; and please let me know in case you have any further concerns.

best regards,

philbucher commented:
@elafnm I am fine with your changes, I removed my blocker

It is up to @rubenzorrilla now

elafnm commented:
thank you very much dear @philbucher,

best regards,

rubenzorrilla commented on kratos/includes/variables.h:
```diff
@@ -412,7 +412,7 @@ namespace Kratos
     //for Structural application:
     KRATOS_DEFINE_VARIABLE( Matrix, INERTIA )
-
+    
```
In the future avoid these kind of aesthetic changes in such a big PR. (probably is something related with the configuration of your code editor). These things make the review process much tedious.

rubenzorrilla commented on kratos/sources/variables.cpp:
```diff
@@ -959,8 +959,10 @@ void KratosApplication::RegisterVariables() {
 
     KRATOS_REGISTER_VARIABLE(INTEGRATION_WEIGHT)
     KRATOS_REGISTER_3D_VARIABLE_WITH_COMPONENTS(INTEGRATION_COORDINATES)
+
```
Same as in `variables.h`

rubenzorrilla reviewed (approved):
Thanks for taking into account our demands. Since all the changes are local in your application I'm approving.

As a comment aside, it's better to do more PRs with less changes. In this way the review will be much faster and you won't need to wait for so long for merging with the master.

elafnm commented:
Thank you very much dears @playpaolo, @rubenzorrilla, @philbucher, @roigcarlo, and @loumalouomega for your supports and approvals,

sure will follow your recommendations for future PRs,

with our best regards,
